internal/js-dependency-scanner/src/main/java/org/wso2/security/tools/scanner/dependency/js/preprocessor/AtuwaDownloader.java

```diff
@@ -0,0 +1,135 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ * /
+ */
+
+package org.wso2.security.tools.scanner.dependency.js.preprocessor;
+
+import org.apache.log4j.Logger;
+import org.wso2.security.tools.scanner.dependency.js.constants.JSScannerConstants;
+import org.wso2.security.tools.scanner.dependency.js.exception.DownloaderException;
+import org.wso2.security.tools.scanner.dependency.js.model.Product;
+import org.wso2.security.tools.scanner.dependency.js.utils.HttpDownloadUtility;
+import org.wso2.security.tools.scanner.dependency.js.utils.UnZipper;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.text.ParseException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Download product pack from Atuwa.
+ * This class extends ResourceDownloader abstract class and implements downloadProductPack method.
+ */
+public class AtuwaDownloader extends ResourceDownloader {
+
+    private static final Logger log = Logger.getLogger(PreProcessor.class);
+    private static String atuwaBaseURl;
+
+    public static void setAtuwaBaseURl(String atuwaBaseURl) {
+        AtuwaDownloader.atuwaBaseURl = atuwaBaseURl;
+    }
+
+    /**
+     * <p>
+     * This is an override method, so javadoc should be visible. This method responsible to download
+     * product packs from atuwa.
+     * </p>
+     * {@inheritDoc}
+     */
+    @Override
+    public List<String> downloadProductPack(Product productDto, String path) throws DownloaderException {
+        return download(path);
+    }
+
+    /**
+     * Download product pack from atuwa url.
+     *
+     * @param path Root directory file path to store the downloaded file.
+     * @return list of downloaded weekly release zip file paths.
+     */
+    private List<String> download(String path) throws DownloaderException {
+        BufferedReader reader = null;
+        List<String> zipFilePathList;
+        String name = null;
+        try {
+            zipFilePathList = new ArrayList<>();
+            String urlString = atuwaBaseURl + File.separator + JSScannerConstants.OB_INDEX_FILE;
+            String downloadUrl;
+            // create the url
+            URL url;
+            try {
+                url = new URL(urlString);
+                // open the url stream, wrap it an a few "readers"
+                reader = new BufferedReader(new InputStreamReader(url.openStream()));
+            } catch (MalformedURLException e) {
+                throw new DownloaderException("Malformed URL has occurred.", e);
```

Comment:
Better to print the URL with the exceptions, so it will be easier to troubleshoot.